spec/outputs/tcp_spec.rb

```diff
@@ -24,6 +24,40 @@
 
   let(:event) { LogStash::Event.new('message' => 'foo bar') }
 
+  describe 'handling obsolete settings for client mode' do
+    [{:name => 'ssl_cert', :replacement => 'ssl_certificate', :sample_value => "certificate_path"},
+     {:name => 'ssl_cacert', :replacement => 'ssl_certificate_authorities', :sample_value => "certificate_path"},
+     {:name => 'ssl_enable', :replacement => 'ssl_enabled', :sample_value => true},
+     {:name => 'ssl_verify', :replacement => 'ssl_client_authentication', :sample_value => 'peer'}].each do | obsolete_setting |
+      context "with obsolete #{obsolete_setting[:name]}" do
+        let (:deprecated_config) do
+          config.merge({obsolete_setting[:name] => obsolete_setting[:sample_value]})
+        end
+
+        it "should raise a config error with the appropriate message" do
+          expect { LogStash::Outputs::Tcp.new(deprecated_config).register }.to raise_error LogStash::ConfigurationError, /The setting `#{obsolete_setting[:name]}` in plugin `tcp` is obsolete and is no longer available. Use '#{obsolete_setting[:replacement]}'/i
+        end
+      end
+    end
+  end
+
+  describe 'handling obsolete settings for server mode' do
```

Comment:
Are we missing some config to construct server vs client mode? I cant see how these are different.